web/src/settings_data.ts

```diff
@@ -220,7 +220,14 @@ export function can_edit_user_group(group_id: number): boolean {
 }
 
 export function user_can_create_user_groups(): boolean {
-    return user_has_permission(realm.realm_user_group_edit_policy);
+    if (page_params.is_spectator) {
+        return false;
+    }
```

Comment:
Should `user_has_permission_for_group_setting` be responsible for doing this spectator check? I'd think yes.